### PR TITLE
Bye bye travis welcome gitlab

### LINKED ISSUE #372: Travis CI migration to Gitlab CI
This repository is using Travis CI.

At the beginning of January 2021, CI using Travis CI projects still on travis-ci.org will no longer work.
To avoid migrating to travis-ci.com which is less friendly with Open Source projects like tango-controls, the Tango Steering Committee decided to migrate the tango-controls git repositories to gitlab.com and to use Gitlab CI for the Continuous Integration.

In order to prepare the migration, CI for this repository should be migrated to gitlab-ci and tested.
To work on that topic before the official migration, one could work on a fork under https://gitlab.com/tango-controls/test_gh_import gitlab group.

### PR DESCRIPTION
**Pull request by t-b**
Close #372 

See https://gitlab.com/t-b/tango-doc/-/merge_requests/370.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# See https://docs.gitlab.com/ce/ci/yaml/README.html for the documentation
+
+package:
+  image: debian:10
```
Comment (mliszcz) [inline .gitlab-ci.yml:4]: Consider just using `python:3.9`. It's buster with python and pip preinstalled. I think even `python:3.9-alpine` could work as `build-documentation.sh` script is POSIX sh, we don't need any other utilities.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  stage: build
+  before_script:
+    - apt update
+    - apt install --yes python3-pip
```
Comment (mliszcz) [inline .gitlab-ci.yml:8]: When used in non-interactive way, apt usually prints a warning:
```
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
```
If you decide to keep installing python by yourself, please consider changing to apt-get.

File: .gitlab-ci.yml (hunk 1 above)
Comment (mliszcz) [reply .gitlab-ci.yml:4]: Note that if you go that way you probably need to override entrypoint as by default python images start the REPL.
```yaml
image:
  name: python:9.3-alpine
  entrypoint: /bin/sh
```

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    - pip3 install -r requirements.txt
+  script:
+    - ./build-documentation.sh
+  artifacts:
+    when: always
+    paths:
+      - sphinx-output.log
+      - build
```
Comment (mliszcz) [inline .gitlab-ci.yml:16]: Can you check how big is the build directory? I'm assuming it's tens or hundreds of megabytes. If yes, consider gzipping that to save some disk space and bandwidth on upload.

**Review by mliszcz — COMMENTED** [review]
Thanks for the PR! I added a few comments, please have a look.

File: .gitlab-ci.yml (hunk 1 above)
Comment (t-b) [reply .gitlab-ci.yml:4]: I would prefer to keep using plain debian. In that way we know all the dependencies.

File: .gitlab-ci.yml (hunk 2 above)
Comment (t-b) [reply .gitlab-ci.yml:8]: Done.

File: .gitlab-ci.yml (hunk 3 above)
Comment (t-b) [reply .gitlab-ci.yml:16]: But if we zip it we loose the funtionality to browse the files directly. Selecting browse from https://gitlab.com/t-b/tango-doc/-/jobs/965595694 leads to https://t-b.gitlab.io/-/tango-doc/-/jobs/965595694/artifacts/build/overview/index.html which allows to quickly look at the result.

The size differences are:
$ du -sh build build.gz
140M    build
102M    build.gz

Comment (mliszcz) [reply .gitlab-ci.yml:16]: Got it. Thanks for checking! Probably pictures contribute a lot to the output size, thus there is no big difference in compressed and uncompressed sizes. What you did is indeed better.

**Review by mliszcz — APPROVED** [review]
LGTM!

**Pull request merged**